### ISSUE TITLE
[Tool Proposal] Implement Compiler Syntax Traceback

### ISSUE THREAD

**Issue by Luna-Klatzer** [Para-C/Para-C#12]
**Summary:**
Implementation of a "Traceback" Syntax System, for proper error logging and information that reports like this:

```bash
    return x ? 233
           ^^^^^^^
<file>:<line> SyntaxError: Invalid conditional expression
```

For the base, this will only contain the line of error and a simple error with an error msg. This will be rather simple for now, due to a lot of additional checking and categorising required for specific errors (This is only momentary though)

**Logic of Error Reporting:**
The simple logic and structure of the error reporting system will base on the antlr4 error-listener, which will go through the items and report either lexer or parser errors, which are then categorised and analysed by the `ErrorAnalyser`, which will return an error-class and message for proper logging. When an error is encountered, compilation will stop, but warnings will still be displayed that were recognised until now.

**Comment by Luna-Klatzer** [comment]
This was implemented with recent commit dff8cb272fb81d13416c319f9ff94f5ae536d824

**Comment by Luna-Klatzer** [comment]
Logic has a slight change, where errors will not stop the compilation and it will attempt to report all valid encountered errors. This also means that syntax errors will be reported over multiple files, without restrictions of a single file. This though does not include compilation, meaning if an error is received in *any*, the compilation will stop and no logical analysis will be started.
